src/devices/r900.c

```diff
@@ -0,0 +1,297 @@
+/** @file
+    Neptune r900 protocol decoder for r900 based flow meters, tested with my water meter.
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+ */
+
+/**
+The device uses PPM encoding,
+- 1 is encoded as 30 us pulse.
+- 0 is encoded as 30 us gap.
+A gap longer than 320 us is considered the end of the transmission.
+The device sends a transmission every xx seconds.
+A transmission starts with a preamble of 0xAA,0xAA,0xAA,0xAB,0x52,0xCC,0xD2
+But, it is "zero" based, so if you add a zero bit to the payload, the preamble
+is: 0x55,0x55,0x55,0x55,0xA9,0x66,0x69,0x65
+It should be sufficient to find the start of the data after 0x55,0x55,0x55,0xA9,0x66,0x69,0x65.
+
+Once the payload is decoded, the message is as follows:
+              (from https://github.com/bemasher/rtlamr/wiki/Protocol#r900-consumption-message)
```

Comment:
use valid markup, don't indent this line. Itemize (dashes in front) the next lines.